host/containerinit/init.go

```diff
@@ -395,6 +400,89 @@ func getCmdPath(c *Config) (string, error) {
 	return cmdPath, nil
 }
 
+func monitor(port host.Port, container *ContainerInit, env map[string]string) (discoverd.Heartbeater, error) {
+	config := port.Service
+	client := discoverd.NewClientWithURL(env["DISCOVERD"])
+
+	if config.Create {
+		// TODO: maybe reuse maybeAddService() from the client
+		if err := client.AddService(config.Name); err != nil {
+			if je, ok := err.(hh.JSONError); !ok || je.Code != hh.ObjectExistsError {
+				return nil, fmt.Errorf("something went wrong with discoverd: %s", err)
+			}
+		}
+	}
+	localAddr := fmt.Sprintf("127.0.0.1:%v", port.Port)
```

Comment:
I ran into problems originally, but I think that might have happened because of other problems. Now that we have tests, I'll give it a try.